air/src/proof.rs

```diff
@@ -86,74 +86,72 @@ impl ExecutionProof {
 
 /// A set of parameters specifying how Miden VM execution proofs are to be generated.
 #[derive(Debug, Clone, Eq, PartialEq)]
-pub struct ProofOptions {
-    pub options: WinterProofOptions,
+pub struct ProvingOptions {
     pub hash_fn: HashFunction,
+    pub exec_options: ExecutionOptions,
+    pub proof_options: WinterProofOptions,
 }
 
-impl ProofOptions {
+impl ProvingOptions {
     // CONSTRUCTOR
     // --------------------------------------------------------------------------------------------
 
-    /// Creates a new instance of [ProofOptions] from the specified parameters.
+    /// Creates a new instance of [ProvingOptions] from the specified parameters.
     pub fn new(
-        num_queries: usize,
-        blowup_factor: usize,
-        grinding_factor: u32,
-        field_extension: FieldExtension,
-        fri_folding_factor: usize,
-        fri_max_remainder_size: usize,
+        proof_options: WinterProofOptions,
+        exec_options: ExecutionOptions,
         hash_fn: HashFunction,
     ) -> Self {
-        let options = WinterProofOptions::new(
-            num_queries,
-            blowup_factor,
-            grinding_factor,
-            field_extension,
-            fri_folding_factor,
-            fri_max_remainder_size,
-        );
-        Self { options, hash_fn }
-    }
-
-    /// Creates a new preset instance of [ProofOptions] targeting 96-bit security level.
+        Self {
+            hash_fn,
+            exec_options,
+            proof_options,
+        }
+    }
+
+    /// Creates a new preset instance of [ProvingOptions] targeting 96-bit security level.
     ///
     /// If `recursive` flag is set to true, proofs will be generated using an arithmetization-
     /// friendly hash function (RPO). Such proofs are well-suited for recursive proof verification,
     /// but may take significantly longer to generate.
-    pub fn with_96_bit_security(recursive: bool) -> Self {
+    pub fn with_96_bit_security(recursive: bool, exec_options: ExecutionOptions) -> Self {
```

Comment:
Let's use a "builder" pattern instead. Meaning, we could add a separate `with_execution_options()` method which would look something like this:
```Rust
pub fn with_execution_options(self, exec_options: ExecutionOptions) -> Self
```
Then, we'd just chain these methods as needed:
```Rust
let options = ProvingOptions::with_96_bit_security(true)
    .with_execution_options(exec_options);
```